aioshelly/wsrpc.py

```diff
@@ -204,13 +268,24 @@ def connected(self) -> bool:
         """Return if we're currently connected."""
         return self._client is not None and not self._client.closed
 
+    def set_auth(self, auth: dict[str, Any]) -> None:
+        """
+        Sets digest auth params
+        """
+        self._auth = auth
+
     async def call(
         self, method: str, params: dict[str, Any] | None = None, timeout: int = 10
     ) -> dict[str, Any]:
         """Websocket RPC call."""
         if self._client is None:
             raise RuntimeError("Not connected")
 
+        if self._auth:
+            if not params:
+                params = {}
+            if "auth" not in params:
```

Comment:
`auth` should not be in params. It should be treated separately. For one, it is not impossible for a method to require an `auth` parameter.